src/pages/api/auth/github/callback.ts

```diff
@@ -0,0 +1,93 @@
+import { APIRoute } from "astro";
+import {
+	fetchGitHubAccessToken,
+	fetchGitHubUser,
+} from "../../../../lib/game-saving/github";
+import {
+	getSession,
+	updateUserGitHubToken,
+} from "../../../../lib/game-saving/account";
+
+export const get: APIRoute = async ({ request, cookies }) => {
+	const url = new URL(request.url);
+	const code = url.searchParams.get("code");
+	const state = url.searchParams.get("state");
+
```

Comment:
Is this endpoint triggered by the GitHub API?

Reply:
yes, it's a callback after user login succesfully after github auth flow.